24H Study Room/ViewModels/LoginViewModel.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  LoginViewModel.swift
+//  24H Study Room
+//
+//  Created by Jia-Jiuan Tsai on 2022-09-05.
+//
+
+import Foundation
+
+class LoginViewModel: ObservableObject {
+    private(set) var fireAuthService: FireAuthServicing
+    @Published var inputEmail = ""
+    @Published var inputPassword = ""
+    @Published var emailIsValid = false
```

Comment:
Duplicated state:

You have already modelled the fact that the email is valid, by the existence of `email`. You can check it with `email != nil`. The property `emailIsValid` is redundant.